src/main/java/duke/TaskList.java

```diff
@@ -8,36 +8,68 @@
 
     private final ArrayList<Task> list;
 
+    /**
+     * Constructs an empty list of tasks.
+     */
     TaskList() {
         this.list = new ArrayList<Task>();
     }
 
+    /**
+     * Constructs a {@code TaskList} object from a list of tasks.
+     * @param list the list of tasks
+     */
     TaskList(ArrayList<Task> list) {
         this.list = list;
     }
 
+    /**
+     * Adds a new task to the list of tasks.
+     * @param t the new task to add
+     * @return a new list of tasks with the new task added
+     */
     public TaskList add(Task t) {
```

Comment:
There's a slight tendency in picking unintuitive names, which is fine in small doses, but can be tiring when its code base with thousands of lines.